nucypher/blockchain/eth/sol/source/contracts/StakingEscrow.sol

```diff
@@ -660,7 +660,7 @@ contract StakingEscrow is Issuer, IERC900History {
             uint256 endIndex = j + numberOfSubStakes;
             require(numberOfSubStakes > 0 && subStakesLength >= endIndex);
             StakerInfo storage info = stakerInfo[staker];
-            require(info.subStakes.length == 0);
+            require(info.subStakes.length == 0 && !info.flags.bitSet(SNAPSHOTS_DISABLED_INDEX));
```

Comment:
What's the rationale for this change?

Reply:
Because we already have code that assumes that snapshots were enabled (in `batchDeposit`) and `setSnapsots` works for everybode (not only staker), I see that we need to protect one or another method